src/Tribe/Template/Month.php

```diff
@@ -421,42 +426,159 @@ protected function set_events_in_month() {
 		}
 
 		/**
-		 * Get the events for a single day
+		 * Retrieves beginning/end times for a given date
 		 *
-		 * @param string $date
+		 * @param string $date Y-m-d date string
+		 * @param string $key Key of cached data to retrieve
 		 *
-		 * @return WP_Query
+		 * return string|int
 		 */
-		private function get_daily_events( $date ) {
+		private function get_cutoff_details( $date, $key ) {
+			static $beginnings_and_ends = array();
 
-			$beginning_of_day           = tribe_event_beginning_of_day( $date );
-			$beginning_of_day_timestamp = strtotime( $beginning_of_day );
-
-			$end_of_day           = tribe_event_end_of_day( $date );
-			$end_of_day_timestamp = strtotime( $end_of_day );
+			if ( empty( $beginnings_and_ends[ $date ] ) ) {
+				$beginnings_and_ends[ $date ] = array(
+					'beginning' => tribe_event_beginning_of_day( $date ),
+					'end' => tribe_event_end_of_day( $date ),
+				);
 
-			$event_ids_on_date = array();
+				$beginnings_and_ends[ $date ]['beginning_timestamp'] = strtotime( $beginnings_and_ends[ $date ]['beginning'] );
+				$beginnings_and_ends[ $date ]['end_timestamp'] = strtotime( $beginnings_and_ends[ $date ]['end'] );
+			}
 
-			// loop through all the events in the month and find the ones on the requested date
-			foreach ( $this->events_in_month as $event ) {
+			return $beginnings_and_ends[ $date ][ $key ];
+		}
 
-				$event_start = strtotime( $event->EventStartDate );
-				$event_end   = strtotime( $event->EventEndDate );
+		/**
+		 * Breaks the possible collection of events down by grid date
+		 *
+		 * @param string $date Y-m-d formatted date to retrieve events for
+		 *
+		 * @return array
+		 */
+		private function get_event_ids_by_day( $date ) {
+			if ( ! $this->event_ids_by_day ) {
+				$this->event_ids_by_day = array();
 
-				// check if the event happens on this day
-				if ( Tribe__Events__Date_Utils::range_coincides( $beginning_of_day_timestamp, $end_of_day_timestamp, $event_start, $event_end ) ) {
-					// check if the event has the term being viewed, if not, skip it
+				// Let's loop over all of the events in the month and assign them to days
+				foreach ( $this->events_in_month as $event ) {
+					// if we're querying by category and the event doesn't have it, skip the event
 					if ( ! empty ( $this->queried_event_cats ) ) {
 						if ( ! has_term( $this->queried_event_cats, Tribe__Events__Main::TAXONOMY, $event ) ) {
 							continue;
 						}
 					}
-					$event_ids_on_date[] = $event->ID;
+
+					$event_start = strtotime( $event->EventStartDate );
+					$event_end   = strtotime( $event->EventEndDate );
```

Comment:
Can we switch to using `tribe_get_start_date()` and `tribe_get_end_date()` when setting these two vars? The benefit is they automatically account for the current timezone setting which is important if the current timezone mode is "sitewide" and some of the events are in a different timezone.